src/Nix/Builtins.hs

```diff
@@ -134,6 +134,7 @@ builtinsList = sequence [
     , add2 Normal   "div"                        div_
     , add2 Normal   "elem"                       elem_
     , add2 Normal   "elemAt"                     elemAt_
+    , add0 Normal   "false"                      (return $ nvConstant $ NBool False)
```

Comment:
I may mistaken, but if you use add', I think just `return False` would work here...

Reply:
Doesn't obviously work:
```
src/Nix/Builtins.hs:137:7: error:
    • Could not deduce (ToBuiltin m (m1 Bool))
        arising from a use of ‘add'’
      from the context: MonadNix e m
        bound by the type signature for:
                   builtinsList :: forall e (m :: * -> *).
                                   MonadNix e m =>
                                   m [Builtin m]
        at src/Nix/Builtins.hs:104:1-59
      The type variable ‘m1’ is ambiguous
      Relevant bindings include
        add' :: forall a.
                ToBuiltin m a =>
                BuiltinType -> Text -> a -> m (Builtin m)
          (bound at src/Nix/Builtins.hs:211:5)
        builtinsList :: m [Builtin m] (bound at src/Nix/Builtins.hs:105:1)
      These potential instances exist:
        instance (MonadNix e m, ToNix a m (NValue m)) =>
                 ToBuiltin m (Prim m a)
          -- Defined at src/Nix/Builtins.hs:896:10
        instance (MonadNix e m, FromNix a m (NValue m), ToBuiltin m b) =>
                 ToBuiltin m (a -> b)
          -- Defined at src/Nix/Builtins.hs:900:10
    • In the expression: add' Normal "false" (return False)
      In the first argument of ‘sequence’, namely
        ‘[do version <- toValue ("2.0" :: Text)
             pure $ Builtin Normal ("nixVersion", version),
          do version <- toValue (5 :: Int)
             pure $ Builtin Normal ("langVersion", version),
          add0 Normal "nixPath" nixPath, add TopLevel "abort" throw_, ....]’
      In the expression:
        sequence
          [do version <- toValue ("2.0" :: Text)
              pure $ Builtin Normal ("nixVersion", version),
           do version <- toValue (5 :: Int)
              pure $ Builtin Normal ("langVersion", version),
           add0 Normal "nixPath" nixPath, add TopLevel "abort" throw_, ....]
    |
137 |     , add' Normal   "false"                      (return False)
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Nix/Builtins.hs:137:51: error:
    • Could not deduce (Monad m1) arising from a use of ‘return’
      from the context: MonadNix e m
        bound by the type signature for:
                   builtinsList :: forall e (m :: * -> *).
                                   MonadNix e m =>
                                   m [Builtin m]
        at src/Nix/Builtins.hs:104:1-59
      The type variable ‘m1’ is ambiguous
      These potential instances exist:
        instance Monad A.Result
          -- Defined in ‘aeson-1.2.4.0:Data.Aeson.Types.Internal’
        instance Monad (Either e) -- Defined in ‘Data.Either’
        instance Monad Dual -- Defined in ‘Data.Monoid’
        ...plus 12 others
        ...plus 61 instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the third argument of ‘add'’, namely ‘(return False)’
      In the expression: add' Normal "false" (return False)
      In the first argument of ‘sequence’, namely
        ‘[do version <- toValue ("2.0" :: Text)
             pure $ Builtin Normal ("nixVersion", version),
          do version <- toValue (5 :: Int)
             pure $ Builtin Normal ("langVersion", version),
          add0 Normal "nixPath" nixPath, add TopLevel "abort" throw_, ....]’
    |
137 |     , add' Normal   "false"                      (return False)
    |                                                   ^^^^^^^^^^^^
```

Using type applications, I get

```
src/Nix/Builtins.hs:137:7: error:
    • Could not deduce (ToBuiltin m (m Bool))
        arising from a use of ‘add'’
      from the context: MonadNix e m
        bound by the type signature for:
                   builtinsList :: forall e (m :: * -> *).
                                   MonadNix e m =>
                                   m [Builtin m]
        at src/Nix/Builtins.hs:104:1-59
    • In the expression: add' Normal "false" (return @m False)
      In the first argument of ‘sequence’, namely
        ‘[do version <- toValue ("2.0" :: Text)
             pure $ Builtin Normal ("nixVersion", version),
          do version <- toValue (5 :: Int)
             pure $ Builtin Normal ("langVersion", version),
          add0 Normal "nixPath" nixPath, add TopLevel "abort" throw_, ....]’
      In the expression:
        sequence
          [do version <- toValue ("2.0" :: Text)
              pure $ Builtin Normal ("nixVersion", version),
           do version <- toValue (5 :: Int)
              pure $ Builtin Normal ("langVersion", version),
           add0 Normal "nixPath" nixPath, add TopLevel "abort" throw_, ....]
    |
137 |     , add' Normal   "false"                      (return @m False)
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```